### PR TITLE
singletons

### PR DESCRIPTION
**Pull request by spangin**

### CODE REVIEW

File: src/main/java/net/helix/pendulum/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java [hunk 1]
```diff
@@ -91,6 +91,15 @@
      */
     private int repairBackoffCounter = 0;
 
+    private static final LatestSolidMilestoneTrackerImpl INSTANCE = new LatestSolidMilestoneTrackerImpl();
+
+    private LatestSolidMilestoneTrackerImpl() {
+    }
+
+    public static LatestSolidMilestoneTrackerImpl getInstance() {
```
Comment (dzhelezov) [inline src/main/java/net/helix/pendulum/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java]: can we return interfaces, not classes here?

Comment (spangin) [reply src/main/java/net/helix/pendulum/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java]: Yes, it's possible, but if exact implementation returns the corresponding instance, it's ok. If we like to use interface, the vars has to be declared as interfaces.

Comment (spangin) [reply src/main/java/net/helix/pendulum/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java]: I can update the corresponding vars declaration. Ok?

**Comment by spangin** [comment]
outdate

**Pull request closed without merging**